src/OSPSuite.Presentation/Presenters/ParameterIdentifications/ParameterIdentificationAnalysisChartPresenter.cs

```diff
@@ -54,11 +53,27 @@ public override void UpdateAnalysisBasedOn(IAnalysable analysable)
          {
             _isMultipleRun = _parameterIdentification.Results.Count > 1;
             UpdateAnalysisBasedOn(_parameterIdentification.Results);
+            setOutputMappingNames();
          }
 
          Refresh();
       }
 
+      private void setOutputMappingNames()
+      {
+         var outputMappingNames = new Cache<string, List<string>>();
+         foreach (var outputMapping in _parameterIdentification.OutputMappings.All)
+         {
+            var outputName = outputMapping.FullOutputPath;
```

Comment:
All of this can be encapulated in a separate object. Dealing with Cache/Ditcionary as main API is, IMO, not the right level of API. Too low. Ok for internal stuff but not to be exosed